### PR TITLE
Replace shell script with javascript 

### PR DESCRIPTION
**Pull request by asimm241**
## Description
Added a javascript to check if git is installed locally or not. If yes, create a .git-info file, if no, throw an error. 

Fixes: #910 

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Checklist
- [] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr or @zone117x for review

### CODE REVIEW

File: git-info-script.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { exec } from 'child_process';
+
+exec('git --version', (error, _stdout, stderr) => {
+  if (error) {
+    throw new Error(`"git is missing", please install git and retry\n${error.message}`);
+  }
+  if (stderr) {
+    throw new Error(`"git is missing", please install git and retry\n${stderr}`);
+  }
+  const gitInfo = `echo \"$(git rev-parse --abbrev-ref HEAD)\n$(git log -1 --pretty=format:%h)\n$(git describe --tags --abbrev=0)\" > ./.git-info`;
```
Comment (zone117x) [inline git-info-script.mjs:10]: If this is using *nix shell exec anyway, there's no point it moving it to a js file. Have you tested if this works on Windows?

File: git-info-script.mjs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import  shell from 'shelljs';
```
Comment (rafaelcr) [inline git-info-script.mjs:1]: Change the name of this file to `git-info-generator.mjs` so intention is clearer

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: git-info-generator.mjs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import  shell from 'shelljs';
+
+if (!shell.which('git')) {
+  throw new Error(`"git is missing", please install git and retry`);
+}
+shell
+  .exec(
+    `echo \"$(git rev-parse --abbrev-ref HEAD)\n$(git log -1 --pretty=format:%h)\n$(git describe --tags --abbrev=0)\"`
```
Comment (zone117x) [inline git-info-generator.mjs:8]: Any reason to be running shell code like this as opposed to using [`child_process.spawnSync(...)`](https://nodejs.org/docs/latest-v16.x/api/child_process.html#child_processspawnsynccommand-args-options)?

Comment (asimm241) [reply git-info-generator.mjs:8]: It is done this way to support it on all platform including windows. Otherwise OS there would be different code for windows like [spawning-bat-and-cmd-files-on-windows](https://nodejs.org/api/child_process.html#spawning-bat-and-cmd-files-on-windows)

Comment (zone117x) [reply git-info-generator.mjs:8]:
```ts
const gitInfo = [
  'git rev-parse --abbrev-ref HEAD',
  'git log -1 --pretty=format:%h',
  'git describe --tags --abbrev=0',
].map(r => execSync(r, { encoding: 'utf8' }));
fs.writeFileSync('.git-info', gitInfo.join('\n'));
```
I think something like this would work just fine

Comment (asimm241) [reply git-info-generator.mjs]: @zone117x What if the user has git installed but does not have tags fetch locally. should we throw an error in that case or ignore it?

Comment (zone117x) [reply git-info-generator.mjs]: Good question, I think if git repo data is missing from disk then this should throw. @rafaelcr thoughts?

Comment (rafaelcr) [reply git-info-generator.mjs]: Yes, that's a very good point @asimm241. IIRC, though, git doesn't return status 1 if the tags are missing from the repo so you'll probably have to check the output message directly.

File: git-info-generator.mjs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { execSync } from 'child_process';
+import { writeFileSync } from 'fs';
+import shell from 'shelljs';
+
+if (!shell.which('git')) {
```
Comment (zone117x) [inline git-info-generator.mjs:5]: No need to pull in the shelljs dependency for this. Something like this should work:
```js
try {
  execSync('git --version');
} catch (error) {
  console.error(error.message);
  throw new Error(`git is missing, please install git and retry`);
}
```

File: tsconfig.json [hunk 5]
```diff
@@ -18,6 +18,6 @@
       "@stacks/blockchain-api-client": ["client/src"]
     }
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "git-info-generator.mjs"],
```
Comment (zone117x) [inline tsconfig.json:21]: What's the reason for this change?

File: git-info-generator.mjs (hunk 4 above)
Comment (asimm241) [reply git-info-generator.mjs:5]: do you think It would work on windows?

File: tsconfig.json (hunk 5 above)
Comment (asimm241) [reply tsconfig.json:21]: oh, that's slipped in, I don't think we need that anymore.

File: .github/workflows/stacks-blockchain-api.yml [hunk 6]
```diff
@@ -75,6 +75,9 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
+        
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
```
Comment (rafaelcr) [inline .github/workflows/stacks-blockchain-api.yml:80]: Instead of this step, add this to the `actions/checkout@v2` step:
```yml
with:
  fetch-depth: 0
```

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: git-info-generator.mjs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { execSync } from 'child_process';
+import { writeFileSync } from 'fs';
+
+try {
+  execSync('git --version');
```
Comment (zone117x) [inline git-info-generator.mjs:5]: Wrapping this and the rest of calls in the same try-catch can show a misleading error message. 

If git repo data isn't available it will show "git not installed' which is incorrect.

**Review by zone117x — APPROVED** [review]
LGTM! We should probably hold off merging this until we cut the next release.

**Review by rafaelcr — APPROVED** [review]
